tools/mkldscript.c

```diff
@@ -51,11 +51,13 @@ static void write_ld_script(FILE *fout)
                   "    ..%s ", seg->name, seg->name, seg->name, seg->name);
 
         if (seg->fields & (1 << STMT_after))
-            fprintf(fout, "_%sSegmentEnd ", seg->after);
+            fprintf(fout, "(_%sSegmentEnd + %i) & ~ %i ", seg->after, seg->align - 1, seg->align - 1);
         else if (seg->fields & (1 << STMT_number))
             fprintf(fout, "0x%02X000000 ", seg->number);
         else if (seg->fields & (1 << STMT_address))
             fprintf(fout, "0x%08X ", seg->address);
+        else
+            fprintf(fout, "ALIGN(0x%X) ", seg->align);
```

Comment:
Probably shouldn't write this at all when align is unset (align(0) doesn't hurt but idk if the align field is initialized to 0. Anyway just check stmt_align)

Reply:
`seg->align` defaults to 16. Is this a bug?

I did try checking if `STMT_align` was set but I thought it made the logic here too complicated. I think writing `ALIGN(0x1)` or whatever in all cases is fine

Reply:
I mean, I think checking for `STMT_align` together with `STMT_address` was too complicated, so I didn't check here to be consistent